### PR TITLE
Cbayer jdk17 spring 3.0.2

### PR DESCRIPTION
**Pull request by chrbayer84**
This resolves #151 #584 #584 #586 #587 #583 #593

### CODE REVIEW

**Comment by chrbayer84** [comment]
tests still failing though. App context doesn't come up.

File: build.gradle [hunk 1]
```diff
@@ -5,14 +5,14 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath("org.springframework.boot:spring-boot-gradle-plugin:2.5.6")
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:3.0.2")
```
Comment (twoversionsofme) [inline build.gradle:8]: Why not add this to the top-level gradle.properties instead?

File: carbonj.service/build.gradle [hunk 2]
```diff
@@ -190,44 +190,51 @@ release {
...
+    testImplementation group: 'junit', name: 'junit', version: '4.13.2'
+    testImplementation group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
+
+    implementation (group:"org.springframework.boot", name:"spring-boot-starter-logging", version:"3.0.2") {
         exclude group: 'org.apache.logging.log4j', module: 'log4j-to-slf4j'
         exclude group: 'org.apache.logging.log4j', module: 'log4j-api'
     }
-    compile group:"org.springframework.boot", name:"spring-boot-starter-web", version:"2.7.5"
-    compile group:"org.springframework.boot", name:"spring-boot-starter-jetty", version:"2.7.5"
+    implementation group:"org.springframework.boot", name:"spring-boot-starter-web", version:"3.0.2"
```
Comment (twoversionsofme) [inline carbonj.service/build.gradle:225]: Shouldn't need a version on dependencies when using the spring boot plugin.

File: carbonj.service/src/main/docker/files/entrypoint.sh [hunk 3]
```diff
@@ -120,5 +120,5 @@ fi
 
 cd /app
 exec java $JAVA_OPTS $JAVA_OPTS_OVERRIDE --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED \
--Dlogback.debug=true -cp /app:/app/lib/* com.demandware.carbonj.service.engine.CarbonJServiceMain \
---spring.config.location=$CONFIG_LOCATIONS
+--add-opens java.base/java.util=ALL-UNNAMED -Dlogback.debug=true -cp /app:/app/lib/* com.demandware.carbonj.service.engine.CarbonJServiceMain \
```
Comment (twoversionsofme) [inline carbonj.service/src/main/docker/files/entrypoint.sh:123]: Any reason for this?

File: carbonj.service/src/main/java/com/demandware/carbonj/service/engine/cfgCarbonJ.java [hunk 4]
```diff
@@ -654,22 +598,17 @@ public ConfigServerUtil configServerUtil(ScheduledExecutorService s, RestTemplat
                 server.setAttribute( "org.eclipse.jetty.server.Request.maxFormContentSize", jettyMaxFormContentSize );
 
                 // TODO change to use logback ch.qos.logback.access.jetty.RequestLogImpl?
-                NCSARequestLog ncsaLog = new NCSARequestLog( jettyLogfilePath );
-                ncsaLog.setExtended( true );
-                ncsaLog.setAppend( true );
-                ncsaLog.setLogTimeZone( "GMT" );
-                ncsaLog.setRetainDays( jettyRequestLogRetentionDays );
+//                NCSARequestLog ncsaLog = new NCSARequestLog( jettyLogfilePath );
```
Comment (twoversionsofme) [inline carbonj.service/src/main/java/com/demandware/carbonj/service/engine/cfgCarbonJ.java:601]: Removing request log?

File: carbonj.service/src/main/java/com/demandware/carbonj/service/engine/kinesis/cfgCheckPointMgr.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    @Value( "${metrics.store.checkPoint.offset.default.mins:5}" ) private int defaultCheckPointOffset;
+
+    @Value( "${metrics.store.checkPoint.provider:filesystem}" ) private String checkPointProvider;
+
+    @Value( "${metrics.store.checkPoint.applicationName:cjajna}" ) private String checkPointApplicationName;
+
+    @Value( "${metrics.store.checkPoint.table.provisioned.throughput:2}" ) private int
+            checkPointTableProvisionedThroughput;
+
+    private static final Logger log = LoggerFactory.getLogger( cfgCheckPointMgr.class );
```
Comment (twoversionsofme) [inline carbonj.service/src/main/java/com/demandware/carbonj/service/engine/kinesis/cfgCheckPointMgr.java:43]: Why not start using @Slf4j annotation which I believe comes with Spring Boot.

File: carbonj.service/build.gradle [hunk 6]
```diff
@@ -190,44 +190,51 @@ release {
...
-    testCompile (group:"org.springframework.boot", name:"spring-boot-starter-test", version:"2.7.5")
+    implementation group:"org.springframework.boot", name:"spring-boot-starter-actuator", version:"3.0.2"
+    testImplementation (group:"org.springframework.boot", name:"spring-boot-starter-test", version:"3.0.2")
 
 }
 
 test {
     maxHeapSize = "2g"
+    jvmArgs = [
+            '--add-opens', 'java.base/java.util=ALL-UNNAMED',
```
Comment (chrbayer84) [inline carbonj.service/build.gradle:236]: If that's removed, one of the test cases around CarbonJEvent logs will fail.

File: carbonj.service/src/main/docker/files/entrypoint.sh (hunk 3 above)
Comment (chrbayer84) [reply carbonj.service/src/main/docker/files/entrypoint.sh:123]: Yes, apparently we are serializing objects from java.lang in CarbonJEvents. If we don't add this, CJ will start spitting out runtime exceptions..

File: carbonj.service/src/main/java/com/demandware/carbonj/service/engine/cfgCarbonJ.java (hunk 4 above)
Comment (chrbayer84) [reply carbonj.service/src/main/java/com/demandware/carbonj/service/engine/cfgCarbonJ.java:601]: Good catch. The class was removed from Jetty. Not sure how to get this functionality back.

File: carbonj.service/src/main/java/com/demandware/carbonj/service/engine/kinesis/cfgCheckPointMgr.java (hunk 5 above)
Comment (chrbayer84) [reply carbonj.service/src/main/java/com/demandware/carbonj/service/engine/kinesis/cfgCheckPointMgr.java:43]: that might be a topic for a different PR..

File: carbonj.service/build.gradle (hunk 2 above)
Comment (chrbayer84) [reply carbonj.service/build.gradle:225]: unfortunately it is needed here.. I think it works for project dependencies but not for buildscript dependencies.

**Review by twoversionsofme — APPROVED** [review]

**Pull request merged**